extensions/wikia/GlobalNavigation/scripts/GlobalNavigationNotifications.js

```diff
@@ -33,89 +33,66 @@ require(
 				this.unreadCount = parseInt(this.$notificationsCount.html(), 10);
 
 				this.$notifications
-					.mouseenter( this.proxy( this.updateCounts ) )
-					.mouseenter( this.proxy( this.fetchForCurrentWiki ) );
+					.mouseenter(this.proxy(this.updateCounts))
+					.mouseenter(this.proxy(this.fetchForCurrentWiki));
 
-				this.$wallNotifications.add( $('#pt-wall-notifications') )
-					.on('click', '.notifications-markasread', this.proxy( this.markAllAsReadAllWikis ));
+				this.$wallNotifications.add($('#pt-wall-notifications'))
+					.on('click', '.notifications-markasread', this.proxy(this.markAllAsReadAllWikis));
```

Comment:
After doing some reading I believe we should leave it as it is. Function.prototype.bind and $.proxy() are not exactly the same. Bind is always returning new function but $.proxy() returns new function only if proxy for this function and for given set of arguments doesn't exist. Notifications code is very fragile and we need to rewrite it anyway when new notification service would be created. Right now, the cost (QA) might be bigger than profit.